for_challenges.py

```diff
@@ -1,31 +1,37 @@
 # Задание 1
 # Необходимо вывести имена всех учеников из списка с новой строки
-
+print("-------- Задание 1. --------")
 names = ['Оля', 'Петя', 'Вася', 'Маша']
-# ???
+for i in names:
+    print(i)
 
 
 # Задание 2
 # Необходимо вывести имена всех учеников из списка, рядом с именем показать количество букв в нём
 # Пример вывода:
 # Оля: 3
 # Петя: 4
-
+print("-------- Задание 2. --------")
 names = ['Оля', 'Петя', 'Вася', 'Маша']
-# ???
+for i in names:
+    print(f"{i}: {len(i)}")
 
 
 # Задание 3
 # Необходимо вывести имена всех учеников из списка, рядом с именем вывести пол ученика
-
+print("-------- Задание 3. --------")
 is_male = {
     'Оля': False,  # если False, то пол женский
     'Петя': True,  # если True, то пол мужской
     'Вася': True,
     'Маша': False,
 }
 names = ['Оля', 'Петя', 'Вася', 'Маша']
-# ???
+for name in names:
+    if is_male[name] is True:
```

Comment:
с логическими штуками просто if достаточно
```suggestion
    if is_male[name]:
```

Reply:
Так же поступаем с проверками на пустую строку, пустой словарь, etc.